### PR TITLE
Icon page pivot for Fabric React and Fabric Core icons

### PR DESCRIPTION
**Pull request by jordandrako**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

- Enable rendering an `Icon` component in the `IconsGrid` icon if list provided should render Fabric React icons
- Add pivot to the Fabric Icons page
- Add `AllIconNames.json` to @uifabric/icons (generated from the fabric subsetter tool)

#### Focus areas to test

(optional)

image

image

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/9560)

### CODE REVIEW

File: apps/fabric-website/src/pages/Styles/FabricIconsPage/FabricIconsPage.tsx [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
 import { IStylesPageProps, StylesAreaPage } from '../StylesAreaPage';
 import { FabricIconsPageProps } from './FabricIconsPage.doc';
 import * as styles from './FabricIconsPage.module.scss';
 import { Platforms } from '../../../interfaces/Platforms';
+import * as IconNames from '@uifabric/icons/src/IconNames';
+
+const allFabricIcons: { name: string }[] = [];
+// IconNames is a const enum, so use the following as a workaround to iterate over it.
+// tslint:ignore-next-line no-any
+for (const iconName in (IconNames as any).IconNames) {
```
Comment (ecraig12345) [inline apps/fabric-website/src/pages/Styles/FabricIconsPage/FabricIconsPage.tsx]: I don't think this will even work at runtime? Const enums are compiled out.

File: packages/icons/.npmignore [hunk 2]
```diff
@@ -21,9 +21,10 @@ node_modules
 src/**/*
 !src/**/*.types.ts
 !src/**/examples/*.tsx
+!src/IconNames.ts
```
Comment (ecraig12345) [inline packages/icons/.npmignore]: Why this change?

Comment (jordandrako) [reply packages/icons/.npmignore]: Testing if it would make a difference, it did not, so I'm exploring other solutions.

File: apps/fabric-website/src/pages/Styles/FabricIconsPage/FabricIconsPage.tsx (hunk 1 above)
Comment (jordandrako) [reply apps/fabric-website/src/pages/Styles/FabricIconsPage/FabricIconsPage.tsx]: Correct. This method worked in vr-tests, but it's not working here at build time (does work when running dev-server, though).

**Comment by jordandrako** [comment]
Closing this until I find a better solution for iterating over all fabric icons.

**Comment by ecraig12345** [comment]
Just to be clear, is the AllIconNames.json auto-generated by default? And have you updated any relevant instructions about copying files (or whatever the process is for generating icon sets)?

**Comment by jordandrako** [comment]
@Jahnp and I talked offline and decided to create a template in the icon subsetter that would generate a json file with all icons, and then we'd copy that to the icons package. I'm not sure where I'd document that process.

Currently, the new template for the subsetter is in PR, so using the live version won't yet get you the file.

**Comment by jordandrako** [comment]
Hold on, there are a few version issues with MDL2 at the moment, hold off on merging.
